src/web/js/trove/d3-lib.js

```diff
@@ -397,70 +410,137 @@
-      image.onload = function () {
-          var opts = {
-              canvas: canvas.node(),
-              image: image
-          };
-          var a = document.createElement('a');
-          a.download = 'sample.png';
-          a.href = drawImage(opts).toDataURL('image/png');
-          a.click();
-
-          // the image's size was doubled everytime we click
-          // the button, so remove all data to prevent this
-          // to happen
-          svgData.remove();
-          canvas.remove();
+  function onSave(detached) {
+    var svgData = detached.append('div').style('display', 'none');
+    var imgsrc = getImageAsURL(detached);
+    var svg = detached.select('svg');
+    var canvas = detached.append('canvas')
+      .style('display', 'none')
+      .attr('width', svg.attr('width'))
+      .attr('height', svg.attr('height'));
+    svgData.html('<img src="' + imgsrc + '">');
+
+    var image = new Image;
+    image.src = imgsrc;
+    image.onload = function () {
+      var opts = {
+        canvas: canvas.node(),
+        image: image
       };
+      var a = document.createElement('a');
+      a.download = 'sample.png';
+      a.href = drawImage(opts).toDataURL('image/png');
+      a.click();
+
+      // somehow the image's size was doubled everytime we click
+      // the button, so remove all data to prevent this
+      // to happen
+      svgData.remove();
+      canvas.remove();
     };
   }
 
-  function callBigBang(detached, dimension, retVal, extra) {
+  function callBigBang(detached, restarter, resizer, windowOptions, dimension, retValFunc, extra) {
+    detached.select('.maing')
+      .append('text')
+      .attr('x', (dimension.marginLeft + dimension.width + dimension.marginRight) / 2)
+      .attr('y', dimension.height + dimension.marginTop + (8 * dimension.marginBottom / 11))
+      .html(libJS.htmlspecialchars(RUNTIME.getField(windowOptions, 'title')))
+      .style({
+        position: 'absolute',
+        'font-size': '8pt',
+        'text-anchor': 'middle'
+      });
+
     detached.selectAll('.overlay').style({
       fill: 'none',
       'pointer-events': 'all',
     });
 
-    RUNTIME.pauseStack(function(restarter) {
-      extra(restarter);
+    if (retValFunc === null) {
+      retValFunc = function(restarter) {
+        imageReturn(detached, restarter, function(x) { return x; });
+      }
+    }
+
+    if (RUNTIME.isPyretFalse(RUNTIME.getField(windowOptions, 'interact'))) {
+      RUNTIME.pauseStack(retValFunc);
+    }
+
+
+    var xscaler = libNum.scaler(
+      dimension.minWindowWidth,
+      dimension.maxWindowWidth,
+      0,
+      1
+    );
+
+    var yscaler = libNum.scaler(
+      dimension.minWindowHeight,
+      dimension.maxWindowHeight,
+      0,
+      1
+    );
+
+    var pauseStack;
+
+    if (RUNTIME.isNothing(restarter)) {
+      pauseStack = RUNTIME.pauseStack;
+    } else {
+      pauseStack = function(cb) { cb(restarter) };
+    }
+
+    pauseStack(function(restarter) {
```

Comment:
@jpolitz @blerner this is very hacky. It is to avoid calling `pauseStack` twice. However, each time a user clicks the resize button, it spawns a new stack frame. Recall that I should use `safeCall`?